query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryServiceImplConfig.java

```diff
@@ -34,16 +28,12 @@ public void setQueryRequestHandlersConfig(List<Config> queryRequestHandlersConfi
     this.queryRequestHandlersConfig = queryRequestHandlersConfig;
   }
 
-  public void setTenantColumnName(String tenantColumnName) {
-    this.tenantColumnName = tenantColumnName;
-  }
-
   public static class RequestHandlerConfig {
 
     private String name;
     private String type;
     private String clientConfig;
-    private Map<String, Object> requestHandlerInfo;
+    private Config requestHandlerInfo;
```

Comment:
Not that the old interface was any better, but would it make sense to make this a real interface rather than passing around an untyped map of config? Not sure where the assumptions about config keys are made, but moving that to as close to the file read as possible makes it less error prone and more debuggable.

Reply:
The keys of this config are assumed in the implementations of `RequestHandler`.
The reason this is a generic untyped config is, each handler could look differently and might have different keys for configs. For example: A pinot RequestHandler vs Druid RequestHandler config could look totally different and might need different set of keys.
Having said that, there could be ways to improve this further. I'll think about this and we can fix in future PRs. Given that I haven't changed the conf files now, this shouldn't be a big regression.

Reply:
Agreed it's not a regression and that it needs to be supported per-handler type. The same place we translate to the appropriate handler, we could translate to the appropriate config. That could fairly easily happen inside the handler implementation too, I just prefer to separate concerns and the quicker a `Config` gets converted to a typed object, the easier it is to trace back issues. Nothing need be done right now, for sure.